### PR TITLE
Add voting feature

### PR DESCRIPTION
**Pull request by YuncongZhou**
can upvote (1) or downvote (0)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -77,6 +77,23 @@ const main = async () => {
     }
     res.status(201).send({ postId: id })
   })
+  //vote posts
+  app.put('/votes/:id', async (req,res)=>{
```
Comment (congqiao) [inline src/index.js]: You are missing a space between `req` and `res`. You should run `yarn run format` and `yarn run lint` before committing.

File: src/index.js [hunk 2]
```diff
@@ -77,6 +77,23 @@ const main = async () => {
...
+    let id
+    try {
+      id = ObjectID.createFromHexString(req.params.id)
+    } catch (err) {
+      res.status(400).end()
+      return
+    }
+    const direction = req.body.direction
+    if (direction){
```
Comment (congqiao) [inline src/index.js]: Use `switch`. This `else` includes everything except `0`, which is misleading.

File: src/index.js [hunk 3]
```diff
@@ -77,6 +77,23 @@ const main = async () => {
...
+      await db.collection('posts').updateOne({ _id: id }, { $inc: { upvote: 1 } })
+    } else {
+      await db.collection('posts').updateOne({ _id: id }, { $inc: { downvote: 1 } })
+    }
+    res.status(200).send({ postId: id.toHexString() })
```
Comment (congqiao) [inline src/index.js]: The client already knows the post ID. Maybe return the whole post object. Also return 201 instead of 200.

**Review by congqiao — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 4]
```diff
@@ -77,6 +77,28 @@ const main = async () => {
...
+        await db.collection('posts').updateOne({ _id: id }, { $inc: { downvote: 1 } })
+        break
+      case 1:
+        await db.collection('posts').updateOne({ _id: id }, { $inc: { upvote: 1 } })
+        break
+      default:
+        res.status(400).end()
+        return
+    }
+    res.status(201).send({ postId: id })
```
Comment (congqiao) [inline src/index.js]: The client already knows the post ID. Maybe return the whole post object.

**Review by congqiao — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 4 above)
Comment (YuncongZhou) [reply src/index.js]: Sure.

File: src/index.js [hunk 5]
```diff
@@ -77,6 +77,29 @@ const main = async () => {
...
+        await db.collection('posts').updateOne({ _id: id }, { $inc: { downvote: 1 } })
+        break
+      case 1:
+        await db.collection('posts').updateOne({ _id: id }, { $inc: { upvote: 1 } })
+        break
+      default:
+        res.status(400).end()
+        return
+    }
+    const postObject = await db.collection('posts').findOne({ _id: id })
```
Comment (congqiao) [inline src/index.js]: Just name it `post`...

**Review by congqiao — CHANGES_REQUESTED** [review]

**Review by congqiao — APPROVED** [review]

**Pull request merged**